app/src/main/java/ch/heigvd/pro/b04/android/Utils/Persistent.java

```diff
@@ -0,0 +1,44 @@
+package ch.heigvd.pro.b04.android.Utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import ch.heigvd.pro.b04.android.R;
+import ch.heigvd.pro.b04.android.Utils.Exceptions.TokenNotSetException;
+
+public final class Persistent {
+    private Persistent() {};
+
+    /**
+     * This function returns the token if it was set previously.
+     * @param context The app context. Needed to access the Shared Preferences
+     * @return The token value
+     * @throws TokenNotSetException is thrown if no token was stored
+     */
+    public static String getStoredTokenOrError(Context context) throws TokenNotSetException {
```

Comment:
Maybe we can use an observer pattern here ? This might work well with a custom LiveData.

Reply:
Yes, I've read about that. I'm just not sure we want to set that up right now. Maybe we can add an issue